lark/lark.py

```diff
@@ -230,8 +234,10 @@ def __init__(self, grammar, **options):
 
             if FS.exists(cache_fn):
                 logger.debug('Loading grammar from cache: %s', cache_fn)
+                for name in (set(LarkOptions._defaults) - _LOAD_ALLOWED_OPTIONS):
+                    options.pop(name, None)
```

Comment:
Wait, what? Why? We should throw an error

Reply:
"This option is incompatible with `cache`"

Reply:
It isn't incompatible. Only `_load` thinks they are incompatible. The options that are thrown away are stuff like `parser` and `start` which are compatible if used with cache, but not when used with Standalone.

Reply:
Ah, okay. I see. I guess anything that is really incompatible would be thrown away earlier, right?

Reply:
I hope? Otherwise this is not a bug in my implementation, but a carry over from before. (In theory the cache is only used if the options are the same, and if not the options are validated by generating a full parser)

Reply:
Okay, that makes sense.